example/kubeconfig/main.tf

```diff
@@ -0,0 +1,46 @@
+terraform {
+  required_providers {
+    plural = {
+      source  = "pluralsh/plural"
+      version = "0.2.25"
+    }
+  }
+}
+
+#####################################
+######### Deprecated method #########
+#####################################
+# provider "plural" {
+#   alias   = "deprecated"
+#   use_cli = true
+# }
+#
+# resource "plural_cluster" "deprecated" {
+#   provider = plural.deprecated
+#   name     = "byok"
+#   protect  = "false"
+#   detach   = true
+#   kubeconfig = {
+#     # It can no longer be sourced from environment variables.
+#     config_path = pathexpand("~/.kube/config")
+#   }
+# }
+
+#####################################
+############ New method #############
+#####################################
+provider "plural" {
+  alias   = "new"
+  use_cli = true
+  kubeconfig = {
```

Comment:
why do you need to assign this with an `=` versus a lot of providers where this would just be:

```tf
provider "plural" {
  kubeconfig {
    ....
  }
}

Reply:
According to the docs blocks should no longer be used. https://developer.hashicorp.com/terraform/plugin/framework/handling-data/blocks:
> Use [nested attribute types](https://developer.hashicorp.com/terraform/plugin/framework/handling-data/attributes#nested-attribute-types) instead of block types for new schema implementations. Block support is mainly for migrating legacy SDK-based providers.

Reply:
gotcha